Source/WebCore/platform/graphics/BitmapImageSource.cpp

```diff
@@ -612,7 +615,8 @@ Expected<Ref<NativeImage>, DecodingStatus> BitmapImageSource::nativeImageAtIndex
 
 Expected<Ref<NativeImage>, DecodingStatus> BitmapImageSource::currentNativeImageForDrawing(SubsamplingLevel subsamplingLevel, const DecodingOptions& options)
 {
-    startAnimation(subsamplingLevel, options);
+    if (isAnimationAllowed())
+        startAnimation(subsamplingLevel, options);
```

Comment:
Why is this change is needed? `ImageFrameAnimator::startAnimation()` will return if

```
    // ImageObserver may disallow animation.
    if (!source->isAnimationAllowed())
        return false;
```